.travis.yml

```diff
@@ -90,21 +90,75 @@ install:
   - export AWESOME_VERSION="${TRAVIS_BRANCH}-g$(git rev-parse --short HEAD)"
   - 'if [ "$TRAVIS_PULL_REQUEST" != false ]; then AWESOME_VERSION="${AWESOME_VERSION}-PR${TRAVIS_PULL_REQUEST}"; fi'
 
+  # Functions for custom Travis folds.
+  - |
+    travis_fold_start() {
+      echo -n "travis_"
+      echo "fold:start:$1"
+      travis_fold_current="$1"
+    }
+    travis_fold_end() {
+      echo -n "travis_"
+      echo "fold:end:$travis_fold_current"
+    }
 script:
   - export CMAKE_ARGS="-DLUA_LIBRARY=${LUALIBRARY} -DLUA_INCLUDE_DIR=${LUAINCLUDE} -D OVERRIDE_VERSION=$AWESOME_VERSION -D DO_COVERAGE=${DO_COVERAGE}"
   - |
     if [ -n "$BUILD_IN_DIR" ]; then
-        SOURCE_DIRECTORY="$PWD"
-        mkdir "$BUILD_IN_DIR"
-        cd "$BUILD_IN_DIR"
-        cmake $CMAKE_ARGS "$SOURCE_DIRECTORY"
+      travis_fold_start "build_in_dir"
+      SOURCE_DIRECTORY="$PWD"
+      mkdir "$BUILD_IN_DIR"
+      cd "$BUILD_IN_DIR"
+      cmake $CMAKE_ARGS "$SOURCE_DIRECTORY"
+      cd -
```

Comment:
Actually... can we somehow ensure that this isn't broken accidentally again later? How about adding `rm Makefile` as the first thing in this block to make sure that the "default magic" will not be used later (together with adding a comment explaining why this is done).

I'd do a PR adding this, but it would conflict with your PR.